mecra.csl

```diff
@@ -0,0 +1,1467 @@
+        <else-if type="patent">
+          <group delimiter=" ">
+            <text variable="authority"/>
+            <text variable="number"/>
+          </group>
+        </else-if>
+        <else-if type="motion_picture" match="any"/>
+        <else>
+          <text variable="genre"/>
+        </else>
+      </choose>
+      <choose>
+        <if type="map">
+          <text variable="scale"/>
+        </if>
+        <else-if type="graphic">
+          <text variable="dimensions"/>
+        </else-if>
+      </choose>
+    </group>
+  </macro>
+  <macro name="description">
+    <group delimiter=", ">
+      <group delimiter=". ">
+        <text macro="interviewer"/>
+        <text variable="medium" text-case="capitalize-first"/>
+      </group>
+      <choose>
+        <if variable="title" match="none"/>
+        <else-if type="thesis speech" match="any"/>
+        <else-if type="patent">
+          <group delimiter=" " suffix=", ">
+            <text variable="authority"/>
+            <text variable="number"/>
+          </group>
+        </else-if>
+        <else-if type="motion_picture" match="any"/>
+        <else>
+          <text variable="genre" text-case="capitalize-first" suffix=","/>
+        </else>
+      </choose>
+      <choose>
+        <if type="map">
+          <text variable="scale"/>
+        </if>
+        <else-if type="graphic">
+          <text variable="dimensions"/>
+        </else-if>
+      </choose>
+    </group>
+  </macro>
+  <macro name="container-title-note">
+    <group delimiter=" ">
+      <choose>
+        <if type="chapter paper-conference" match="any">
+          <text term="in"/>
+        </if>
+      </choose>
+      <choose>
+        <if type="webpage">
+          <text variable="container-title" font-style="italic"/>
+        </if>
+        <else-if type="post-weblog">
+          <text variable="container-title" font-style="italic" suffix=" (blog)"/>
+        </else-if>
+        <else-if type="bill legislation legal_case" match="none">
+          <text variable="container-title" font-style="italic"/>
+        </else-if>
+      </choose>
+    </group>
+  </macro>
+  <macro name="container-title">
+    <group delimiter=" ">
+      <choose>
+        <if type="chapter paper-conference" match="any">
+          <choose>
+            <if match="any" variable="translator">
+              <text term="in" text-case="lowercase"/>
+            </if>
+            <else>
+              <text term="in" text-case="capitalize-first"/>
+            </else>
+          </choose>
+        </if>
+      </choose>
+      <choose>
+        <if type="webpage">
+          <text variable="container-title" font-style="italic" suffix=","/>
+        </if>
+        <else-if type="post-weblog">
+          <text variable="container-title" font-style="italic" suffix=" (blog)"/>
+        </else-if>
+        <else-if type="entry-encyclopedia entry-dictionary chapter" match="any">
```

Comment:
move this up into line 472. No need for the separate tests for identical statement.